arangod/ClusterEngine/ClusterIndex.cpp

```diff
@@ -123,6 +130,57 @@ void ClusterIndex::toVelocyPack(
     builder.add(StaticStrings::IndexEstimates, VPackValue(false));
   }
 
+  if (Index::hasFlag(flags, Index::Serialize::Figures)) {
+    uint64_t progress = 0;
+    uint64_t success = 0;
+    auto const shards = _collection.shardIds();
+    auto const body = VPackBuffer<uint8_t>();
+    auto* pool =
+        _collection.vocbase().server().getFeature<NetworkFeature>().pool();
+    std::vector<Future<network::Response>> futures;
+    futures.reserve(shards->size());
+    std::string const prefix = "/_api/index/";
+    for (auto const& shard : *shards) {
+      network::RequestOptions reqOpts;
+      std::string const url =
+          prefix + shard.first + "/" + std::to_string(_iid.id());
+      futures.emplace_back(
+          network::sendRequestRetry(pool, "shard:" + shard.first,
+                                    fuerte::RestVerb::Get, url, body, reqOpts));
+    }
+    for (Future<network::Response>& f : futures) {
+      network::Response const& r = f.get();
+      if (r.fail()) {
+        LOG_TOPIC("afde4", INFO, Logger::CLUSTER)
+            << "Communication error while collecting figures for collection "
+            << _collection.name() + " from " + r.destination;
+      }
```

Comment:
we probably should not go on here if `r` contains a failure.
```suggestion
        continue;
      }
```

Reply:
Note: this and the following error checking sequence can be simplified by checking `r.combinedResult()`, which will already contain the error message in case something went wrong.